go/arrow/flight/flightsql/client.go

```diff
@@ -218,6 +218,54 @@ func (c *Client) ExecuteSubstraitUpdate(ctx context.Context, plan SubstraitPlan,
 	return updateResult.GetRecordCount(), nil
 }
 
+// ExecuteIngest is for executing a bulk ingestion and only returns the number of affected rows.
+func (c *Client) ExecuteIngest(ctx context.Context, rdr array.RecordReader, reqOptions *ExecuteIngestOpts, opts ...grpc.CallOption) (n int64, err error) {
+	var (
+		desc         *flight.FlightDescriptor
+		stream       pb.FlightService_DoPutClient
+		wr           *flight.Writer
+		res          *pb.PutResult
+		updateResult pb.DoPutUpdateResult
+	)
+
+	cmd := (*pb.CommandStatementIngest)(reqOptions)
+	if desc, err = descForCommand(cmd); err != nil {
+		return
+	}
+
+	if stream, err = c.Client.DoPut(ctx, opts...); err != nil {
+		return
+	}
+
+	wr = flight.NewRecordWriter(stream, ipc.WithAllocator(c.Alloc), ipc.WithSchema(rdr.Schema()))
+	wr.SetFlightDescriptor(desc)
+
```

Comment:
don't we need `defer wr.Close()` ?

Reply:
Good catch, forgot to do so. Just added.